cool-core/src/main/java/com/nus/cool/core/cohort/refactor/storage/Scope.java

```diff
@@ -14,7 +17,11 @@ public Scope(Integer l, Integer r){
 
     public Boolean IsInScope(Integer i){
         return i < this.right && i >= this.left;
-    }  
+    }
+
+    public Boolean IsIntersection(Scope scope){
+        return !(scope.getLeft() > right || scope.getRight() < left);
```

Comment:
pay attention [left, right)
Wrong logic.
scope.getLeft() >= this.right || scope.getRight() <= this.left